streams/src/main/java/org/apache/kafka/streams/kstream/internals/foreignkeyjoin/SubscriptionResolverJoinProcessorSupplier.java

```diff
@@ -0,0 +1,106 @@
+import org.apache.kafka.streams.processor.ProcessorSupplier;
+import org.apache.kafka.streams.processor.StateStore;
+import org.apache.kafka.streams.processor.internals.ProcessorContextImpl;
+import org.apache.kafka.streams.state.KeyValueStore;
+import org.apache.kafka.streams.state.SessionStore;
+import org.apache.kafka.streams.state.TimestampedKeyValueStore;
+import org.apache.kafka.streams.state.TimestampedWindowStore;
+import org.apache.kafka.streams.state.WindowStore;
+
+import java.security.NoSuchAlgorithmException;
+import java.util.Random;
+
+public class SubscriptionResolverJoinProcessorSupplier<K, V, VO, VR> implements ProcessorSupplier<K, SubscriptionResponseWrapper<VO>> {
+    private final String stateStoreName;
+    private final Serializer<V> valueSerializer;
+    private final ValueJoiner<V, VO, VR> joiner;
+
+    public SubscriptionResolverJoinProcessorSupplier(final String stateStoreName,
+                                                     final Serializer<V> valueSerializer,
+                                                     final ValueJoiner<V, VO, VR> joiner) {
+        this.stateStoreName = stateStoreName;
+        this.valueSerializer = valueSerializer;
+        this.joiner = joiner;
+    }
+
+    @Override
+    public Processor<K, SubscriptionResponseWrapper<VO>> get() {
+        return new AbstractProcessor<K, SubscriptionResponseWrapper<VO>>() {
+
+            //private KeyValueStore<K, V> originalSource;
+            //TODO - How to handle the other various store types?
+            private TimestampedKeyValueStore<K,V> originalSource;
+
+            @Override
+            public void init(final ProcessorContext context) {
+                super.init(context);
+                originalSource = (TimestampedKeyValueStore<K,V>)context.getStateStore(stateStoreName);
+            }
+
+            @Override
+            public void process(K key, SubscriptionResponseWrapper<VO> value) {
+
+//                final StateStore store = stateManager.getStore(name);
+//                if (store instanceof TimestampedKeyValueStore) {
+//                    return new ProcessorContextImpl.TimestampedKeyValueStoreReadWriteDecorator((TimestampedKeyValueStore) store);
+//                } else if (store instanceof KeyValueStore) {
+//                    return new ProcessorContextImpl.KeyValueStoreReadWriteDecorator((KeyValueStore) store);
+//                } else if (store instanceof TimestampedWindowStore) {
+//                    return new ProcessorContextImpl.TimestampedWindowStoreReadWriteDecorator((TimestampedWindowStore) store);
+//                } else if (store instanceof WindowStore) {
+//                    return new ProcessorContextImpl.WindowStoreReadWriteDecorator((WindowStore) store);
+//                } else if (store instanceof SessionStore) {
+//                    return new ProcessorContextImpl.SessionStoreReadWriteDecorator((SessionStore) store);
+//                }
```

Comment:
Am I going to need to handle all of these statestore types? I suspect I'll need to handle both `TimestampedKeyValueStore` and regular `KeyValueStore`, but given that `context.getStateStore` can return any of them, I don't know if I should expect to handle the remainder.

Reply:
Overcome by events. No longer relevant 👍